src/state/slices/addressBookSlice/selectors.ts

```diff
@@ -0,0 +1,65 @@
+import { createSelector } from '@reduxjs/toolkit'
+import type { ChainId, ChainNamespace } from '@shapeshiftoss/caip'
+import { fromChainId } from '@shapeshiftoss/caip'
+import { matchSorter } from 'match-sorter'
+import createCachedSelector from 're-reselect'
+
+import { addressBookSlice } from './addressBookSlice'
+
+import type { ReduxState } from '@/state/reducer'
+
+export const selectAddressBookEntries = createSelector(
+  addressBookSlice.selectors.selectAddressBookEntries,
+  entries => entries,
+)
+
+export const selectAddressBookEntriesByChainNamespace = createCachedSelector(
+  addressBookSlice.selectors.selectAddressBookEntries,
+  (_state: ReduxState, chainId: ChainId) => chainId,
+  (entries, chainId) => {
+    if (!chainId) return []
+
+    const { chainNamespace } = fromChainId(chainId)
+
+    // Filter entries that match the same chain namespace
+    // This allows addresses to work across all chains of the same type
+    // (e.g., an Ethereum address works on all EVM chains)
```

Comment:
Doesn't generalise to UTXOs however:

```typescript
export const CHAIN_NAMESPACE = {
  Evm: 'eip155',
  Utxo: 'bip122',
  CosmosSdk: 'cosmos',
  Solana: 'solana',
} as const
```

i.e with this logic, a THOR address would be valid for Maya and CosmosHub, a DOGE addy would be valid for BTC etc.